### PR TITLE
Resize pane

### PR DESCRIPTION
**Pull request by ernesto28**
### Requirements

Atom    : 1.23.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

### Description of the Change

Atom gets into a perplexing state for some people (for example) whether by some combination of keys or package actions where there is only one pane, but it does not take up the full width of the editor window. This changes in this Pull request prevent atom to get into this state.

### Alternate Designs

NA

### Why Should This Be In Core?

Fixes a bad behavior in atom UI

### Benefits

Added test for the new code
Better experience in atom UI

### Possible Drawbacks

None that I can think of.

### Applicable Issues

It's applicable to #11673

### CODE REVIEW

File: spec/pane-container-element-spec.coffee [hunk 1]
```diff
@@ -237,3 +237,30 @@ describe "PaneContainerElement", ->
...
+
+  describe "one pane resizing", ->
+    [singlePane] = []
+
+    beforeEach ->
+      container = new PaneContainer(params)
+      singlePane = container.getActivePane()
+
+    describe "when pane:increase-size is triggered", ->
+      it "increases the size of the pane", ->
```
Comment (50Wliu) [inline spec/pane-container-element-spec.coffee]: These spec descriptions are incorrect - can you change them?

File: src/pane.js [hunk 2]
```diff
@@ -155,9 +155,17 @@ class Pane {
 
   getFlexScale () { return this.flexScale }
 
-  increaseSize () { this.setFlexScale(this.getFlexScale() * 1.1) }
+  increaseSize () {
+    if (this.getContainer().getPanes().length > 1){
```
Comment (50Wliu) [inline src/pane.js]: Style: space before opening bracket

File: src/pane.js [hunk 3]
```diff
@@ -155,9 +155,17 @@ class Pane {
...
+      this.setFlexScale(this.getFlexScale() * 1.1)
+    }
+  }
 
-  decreaseSize () { this.setFlexScale(this.getFlexScale() / 1.1) }
+  decreaseSize () {
+    if (this.getContainer().getPanes().length > 1){
```
Comment (50Wliu) [inline src/pane.js]: Style: space before opening bracket

File: spec/pane-container-element-spec.coffee [hunk 4]
```diff
@@ -237,3 +237,30 @@ describe "PaneContainerElement", ->
...
+        expect(singlePane.getFlexScale()).toBe 1
+
+        atom.commands.dispatch(singlePane.getElement(), 'pane:increase-size')
+        expect(singlePane.getFlexScale()).toBe 1
+
+        atom.commands.dispatch(singlePane.getElement(), 'pane:increase-size')
+        expect(singlePane.getFlexScale()).toBe 1
+
+    describe "when pane:decrease-size is triggered", ->
+      it "decreases the size of the pane", ->
```
Comment (50Wliu) [inline spec/pane-container-element-spec.coffee]: Same here

File: spec/pane-container-element-spec.coffee [hunk 5]
```diff
@@ -237,3 +237,30 @@ describe "PaneContainerElement", ->
...
+        expect(singlePane.getFlexScale()).toBe 1
+
+        atom.commands.dispatch(singlePane.getElement(), 'pane:decrease-size')
+        expect(singlePane.getFlexScale()).toBe 1
+
+        atom.commands.dispatch(singlePane.getElement(), 'pane:decrease-size')
+        expect(singlePane.getFlexScale()).toBe 1
```
Comment (50Wliu) [inline spec/pane-container-element-spec.coffee]: Newline should be added at the end of the file

File: spec/pane-container-element-spec.coffee [hunk 6]
```diff
@@ -237,3 +237,30 @@ describe "PaneContainerElement", ->
         atom.commands.dispatch(rightPane.getElement(), 'pane:decrease-size')
         expect(leftPane.getFlexScale()).toBe 1/1.1
         expect(rightPane.getFlexScale()).toBe 1/1.1
+
+  describe "one pane resizing", ->
```
Comment (50Wliu) [inline spec/pane-container-element-spec.coffee]: How about `when only a single pane is present`

**Review by winstliu — CHANGES_REQUESTED** [review]
Thanks for taking this on!  I've left some minor review comments below.

**Comment by ernesto28** [comment]
@50Wliu Thanks, I'll correct them

**Comment by 50Wliu** [comment]
Thanks!  You'll also need to remove the package-lock.json files you checked in as Atom won't build if they're present.

**Comment by ernesto28** [comment]
Got it, I'll delete them

**Review by winstliu — APPROVED** [review]

**Comment by as-cii** [comment]
Looks great, nice work! ⚡️

**Pull request merged**